src/tools/rbd/action/Bench.cc

```diff
@@ -317,18 +349,28 @@ int do_bench(librbd::Image& image, io_type_t io_type,
 
     // Set the thread_offsets of next I/O
     for (i = 0; i < io_threads; ++i) {
-      if (random) {
+      switch (io_pattern) {
+      case IO_PATTERN_RAND:
         thread_offset[i] = (rand() % (size / io_size)) * io_size;
         continue;
+      case IO_PATTERN_SEQ:
+        if (off < (io_size * unit_len * io_threads) ) {
+          thread_offset[i] += io_size;
+        } else {
+          // thread_offset is adjusted to the chunks unassigned to threads.
+          thread_offset[i] = off + (i * io_size);
+        }
+        break;
+      case IO_PATTERN_FULL_SEQ:
+        thread_offset[i] += unit_len;
+        if (thread_offset[i] >= size) {
+          thread_offset[i] = i * io_size;
+        }
+        break;
+      default:
+        break;
       }
-      if (off < (io_size * unit_len * io_threads) ) {
-        thread_offset[i] += io_size;
-      } else {
-        // thread_offset is adjusted to the chunks unassigned to threads.
-        thread_offset[i] = off + (i * io_size);
-      }
-      if (thread_offset[i] + io_size > size)
-        thread_offset[i] = unit_len * i * io_size;
```

Comment:
I think you still need this `if (thread_offset[i] + io_size > size)` check to avoid out of image boundary io. Running it with `--io-total` more than image size shows the issue.

Reply:
Oops -- didn't mean to drop that for the seq case. Thx.